datafusion/core/src/physical_optimizer/enforce_sorting.rs

```diff
@@ -857,14 +852,14 @@ mod tests {
     #[tokio::test]
     async fn test_remove_unnecessary_sort1() -> Result<()> {
         let schema = create_test_schema()?;
-        let source = memory_exec(&schema);
         let sort_exprs = vec![sort_expr("nullable_col", &schema)];
-        let sort = sort_exec(sort_exprs.clone(), source);
-        let spm = sort_preserving_merge_exec(sort_exprs, sort);
+        let physical_plan = TestPlanBuilder::new_memory_exec(&schema)
+            .sort_by_exprs(sort_exprs.clone())
+            .sort_preserving_merge(sort_exprs.clone())
+            .sort_by_exprs(sort_exprs.clone())
+            .sort_preserving_merge(sort_exprs)
+            .build();
 
-        let sort_exprs = vec![sort_expr("nullable_col", &schema)];
-        let sort = sort_exec(sort_exprs.clone(), spm);
-        let physical_plan = sort_preserving_merge_exec(sort_exprs, sort);
         let expected_input = [
             "SortPreservingMergeExec: [nullable_col@0 ASC]",
```

Comment:
The tests already have an exepcted output that nicely confirms the created plan still has the same structure as the previous tests